src/Http/Controllers/Controller.php

```diff
@@ -116,32 +116,44 @@ public function insertUpdateData($request, $slug, $rows, $data)
         return $data;
     }
 
-    public function validateBread($request, $data)
+    public function validateBread($request, $data, $action = null)
     {
         $rules = [];
         $messages = [];
+        $customAttributes = [];
 
         foreach ($data as $row) {
+            if ($row->display_name) $customAttributes[$row->field] = $row->display_name;
+
             $options = json_decode($row->details);
 
             if (isset($options->validation)) {
-                if (isset($options->validation->rule)) {
-                    if (!is_array($options->validation->rule)) {
-                        $rules[$row->display_name] = explode('|', $options->validation->rule);
+                $validationOptions = $options->validation;
+
+                // if store or update-specific validation is defined, use that instead
+                if ($action === 'store' && isset($options->validation->store)) {
+                    $validationOptions = $options->validation->store;
+                } elseif ($action === 'update' && isset($options->validation->update)) {
+                    $validationOptions = $options->validation->update;
+                }
+
+                if (isset($validationOptions->rule)) {
```

Comment:
Try to avoid nested IFs. If you're inside a loop, you can call "continue" to jump to the next loop. This is for better readability and simplicity in the code.